src/conversions.jl

```diff
@@ -317,17 +317,20 @@ function convert_arguments(SL::SurfaceLike, x::AbstractVector{<:Number}, y::Abst
         error("x, y and z need to have the same length. Lengths are $(length.((x, y, z)))")
     end
 
+    # Convert to Float32 before checking for uniqueness
+    x, y, z = map(el32convert, (x, y, z))
```

Comment:
I think this doesn't need to include z, right? Every element from z will be converted to Float32 anyway when assigning it to zs later.

Reply:
Indeed!